### ISSUE TITLE
Moving/copying items into a folder is hard if the folder pane contains no empty space

### ISSUE THREAD

**Issue by LinuxOnTheDesktop** [linuxmint/nemo#2519]
```
 * Nemo version (nemo --version) 4.6.5. On Cinnamon.
 * Is issue with desktop or windowed nemo? Both
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) Mint 20
 * Graphics hardware *and* driver used: Intel UHD Graphics 620; Mesa
 * 32 or 64 bit: 64
 ```

**Issue**

Sometimes the user wants to copy/move something by dragging it (from the desktop or a Nemo window) to a location shown in a (/another) Nemo window. One can do this - or try to do it - in various ways, some of which are problematic.

1) _One drags the item to the icon for the target directory_ (the directory that one wants the item to go in). That works and, if one's 'Places' is in 'tree view', then it is easy to do.

2) _One drags the item into a pane displaying the contents of the target directory_. If there is empty space showing in the pane, then one drags to there and again all is fine. If there is no such space, one has a problem. Indeed one is liable to copy/move the item into a subfolder of one's target, by accident. Still, perhaps I miss something. I find that trying to use the keyboard - ctrl-X followed by ctrl-v - encounters essentially the same problem.

3) _The user might try dragging to the Nemo address bar_. This has a counter-intuitive result. To wit: only *text* gets moved.

**Steps to reproduce**

Please see above. I can provide a video if necessary.

**Expected behaviour**

I am unsure how *2* should be solved. How do other desktop environments / file managers do it?
3: I think the expected behaviour (which I intimate above) should be implemented.

**Additional information**

In dealing with the issue described I encountered two other problems.

I) _I could not recycle a large directory_ (i.e. one that had has many MBs of items in it) on the desktop. The operation failed silently. Nor even could I delete the directory, i.e. bypass the bin (but still using the GUI). _Several copies of the directory were already in the Bin_. But if that was the cause of the problem then the user should be told; the failure should not be silent. Also, though: for some reason - even before I emptied the bin - the problem disappeared.

II) _When trying to copy and move items from the desktop to a Nemo window, the icon on the desktop_ - the icon for the item being operated upon - _behaved strangely_: it disappeared and then reappeared; at once point it jumped from one position on the desktop to another. I have not managed to reproduce the problem. (But cf. #2474.)

**Comment by Jeremy7701** [comment]
Assuming icon display:-
Copy/move files into a pane - or a second window - this works well (see video).
Either drag/drop or ctrl-X + ctrl-V.
[out.ogv.tar.gz](https://github.com/linuxmint/nemo/files/5541166/out.ogv.tar.gz)

Whilst producing this video, I did a dry run - so I now have exact duplicates of the 3 files in the rubbish bin,
Emptying the rubbish bin worked as normal.

**Comment by LinuxOnTheDesktop** [comment]
Thank you, Jeremy. Yet, compare the video I attach, which shows what happens if (1) one's Nemo window is full of *directories* and (2) one is in the *details* view. Indeed, if condition _1_ is met, being in icon view is not that much better.

[out.ogv.tar.gz](https://github.com/linuxmint/nemo/files/5557396/out.ogv.tar.gz)

**Comment by dfrein** [comment]
I can confirm: it is very hard an there is not sufficient visual feedback where to drop files in the detailed view: If the target folder is full of subfolders or compressed archives (e.g. .zip files), nemo will drop the files into the archives or the subfolders under the mouse cursor, even when that was not intended. As a workaround I usually try to drop files in such cases to nemo's title bar, column headers or the path entry widget, but that is not working in most of the cases.

Furthermore there is in my opinion too less visual feedback that a folder or an archive is selected as drop target, I would expect that there is some strong highlighting of the active entry (e.g. full line is highlighted instead of just thin lines surrounding the selected entry.

In other desktop environments that has been solved in a more intuitive way, windows explorer will only drop files to directories if dropped on the filename column, an not into the detail columns. In dolphin, not only filename column, but only the filename itself serves as a drop target.

In nemo's other views it is at least possible to drop files in between of the icons, which is at least the expected behaviour.

**Comment by MarkJeronimus** [comment]
I can't believe this hasn't been addressed in 5 years.

Because of this bug I regularly experience **DATA LOSS!**

When I drag files to the other pane, and notice there's just files below the listed directories, and drop there, I often accidentally drop it on a zip file, and the zip file gets irreversibly changed. (Sure I can go into the zip and delete the files (if it's not too many of them) but he zip will not be binary equal to the original)

I see several reasonable solutions:

- Ask for conformation when dropping (prevents accidental data loss) (Total commander works this way)
- Dropping on the empty space _horizontally_ between the file name and other columns (in List View)

Additionally I see some _acceptable_ solutions:

- Add shortcuts for "Copy to other pane" and "Move to other pane"
- Just don't allow dropping into zip files. It's not that hard to open File Roller and drag files there.
- Allow dropping on the column headers (in List View). I don't see this mentioned explicitly in this thread.

**Comment by mtwebster** [comment]
What version are you using? This has been fixed since Nemo 6.0:
https://github.com/linuxmint/nemo/commit/b9d396e153951b1925e99105857f47d3261a06cb

Sorry this report was never tagged...

**Comment by MarkJeronimus** [comment]
Yes, I'm stuck at Nemo 5.2.4 because an outdated distro that can't be upgraded due to company reasons
